substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/SubstrateNodeLLVMBuilder.java

```diff
@@ -59,6 +81,75 @@ public void emitCGlobalDataLoadAddress(CGlobalDataLoadAddressNode node) {
     @Override
     public Variable emitReadReturnAddress() {
         LLVMValueRef returnAddress = getLIRGeneratorTool().getBuilder().buildReturnAddress(getLIRGeneratorTool().getBuilder().constantInt(0));
-        return new LLVMUtils.LLVMVariable(returnAddress);
+        return new LLVMVariable(returnAddress);
+    }
+
+    private SubstrateLLVMGenerator getGenerator() {
+        return (SubstrateLLVMGenerator) gen;
+    }
+
+    @Override
+    protected int getParamIndex(int index) {
+        return (getGenerator().isEntryPoint() ? 0 : LLVMFeature.SPECIAL_REGISTER_COUNT) + super.getParamIndex(index);
```

Comment:
Is the IsolateThread always at position 0? Also, what happens with entry points that accept the `Isolate` instead of the `IsolateThread`.

Reply:
This is just to offset the parameters if there are special registers passed as parameters as well. The function is maybe not too clear, `(getGenerator().isEntryPoint() ? 0 : LLVMFeature.SPECIAL_REGISTER_COUNT)` is just computing an offset to be added to the actual parameter number, I'll put this in an intermediary variable `offset` to clarify